services/ux-backend/handlers/onboardingtokens/handler.go

```diff
@@ -52,3 +58,40 @@ func handleUnsupportedMethod(w http.ResponseWriter, r *http.Request) {
 		klog.Errorf("failed write data to response writer: %v", err)
 	}
 }
+
+func getAllocatedQuotaInGib(r *http.Request) (*int, error) {
+	var quota = struct {
+		StorageQuota int    `json:"storageQuota"`
+		Unit         string `json:"unit"`
+	}{}
+	err := json.NewDecoder(r.Body).Decode(&quota)
+	if err != nil {
+		// When error is EOF that means request body is empty and quota
+		// is unlimited
+		if err.Error() == "EOF" {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	if quota.StorageQuota <= 0 {
+		return nil, fmt.Errorf("invalid StorageQuota value sent in request body: %v", quota.StorageQuota)
+	}
+
+	if quota.Unit == "" {
+		return nil, fmt.Errorf("invalid Unit type sent in request body, Valid types are [Gi,Ti,Pi]: %v", quota.Unit)
+	}
+	var quotaInGib int
+	switch quota.Unit {
+	case "Gi":
+		quotaInGib = quota.StorageQuota * 1
+	case "Ti":
+		quotaInGib = quota.StorageQuota * 1024
+	case "Pi":
+		quotaInGib = quota.StorageQuota * 1024 * 1024
+	default:
+		return nil, fmt.Errorf("invalid Unit type sent in request body, Valid types are [Gi,Ti,Pi]: %v", quota.Unit)
+	}
```

Comment:
wouldn't a fixed map be better here?
So you will have:

unitAsGiB, ok := UnitToGib[qouta.Unit] 
if !ok { 
    handle error
}
return toPtr(quota.Value * unitAsGiB)